### PR TITLE
Chat tasks: live ring for headless turns, deleted-task notice, header title parity

### PR DESCRIPTION
**Pull request by AkshilVT**
## Summary
- **Ring stuck on done**: chats sent from the app run `claude -p`, which never registers in `~/.claude/sessions`, so `tasks_watch` never watched their transcripts. Now a session named by `history.jsonl` is watched for 10 min. The topbar ring also honors the page's own `running` so it flips at send time.
- **Deleted task ghost row / blank chat**: `EraseTaskModal` broadcasts `tasks-changed` so every list reloads immediately; header drops its cached seed; history endpoint returns `deleted: true` for a tombstoned session and the chat shows an error row instead of a blank conversation.
- **Header/row title mismatch**: with "Title a task by its last message" on, `TaskPeekWho` prints the same `cardTitleLine` as the row.
- **Stale "wrong" title**: `_absorb` drops the kept reply when a newer prompt lands (append-only file, no timestamp race); `_prompt` skips `isSidechain`.

## Test plan
- [x] tsc, `bun test` (5992), pytest touched suites (299) green; 9 new tests
- [x] cmux QA: ring in_progress at t=0 and live on landing without reload; title parity 2/2; delete removes row instantly, deleted URL shows notice; no console errors; no clipping at 380px

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Cross-cutting changes to task polling, history API semantics, and transcript caching affect open-session and erase flows; behavior is covered by new tests but several listing edge cases were touched.
> 
> **Overview**
> Fixes three chat/task UX gaps: **live status**, **erased tasks**, and **consistent titles**.
> 
> **`-p` turns looked idle:** `tasks_watch` now tracks sessions named in `history.jsonl` for ~10 minutes (no registry row), bumps when transcripts settle, and the chat topbar passes **`running`** into `TaskPeekWho` so the ring shows in-progress before the listing poll catches up.
> 
> **Erased tasks opened blank chats:** History responses for tombstoned sessions get **`deleted: true`** (server checks `tasks_store.erased`, not soft-delete). `openSession` / `refreshHistory` route that through **`landDeleted`**: clear optional **`historyCache.delete`**, show an error instead of an empty transcript, and skip caching. Erase flow **`announceTasksChanged()`** after `onDone` so lists reload immediately, plus **`forgetSessionSeed`** so Recent rows do not stick around.
> 
> **Title / `last_message` correctness:** Shared **`useTaskHeadline`** (`cardTitleLine` + pref) aligns peek, modal, and chat header with list rows. Listing scan drops replies when a newer prompt arrives (append order, not timestamps) and ignores **`isSidechain`** prompts/replies.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 62214de5de56308bcf6249a32ca7282408f94de2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: fused_render/tasks_watch.py [hunk 1]
```diff
@@ -381,6 +393,9 @@ def tick() -> set[str]:
     announces nothing — the page's first full listing already has it all."""
     global _primed
     keys = _read_history_tail()
+    now = time.time()
+    for sid in keys:
+        _prompted[sid] = now
```
[the remark this answers is not in the document]
Comment (AkshilVT) [reply fused_render/tasks_watch.py:454]: Fixed in c22f4955f.

File: frontend/src/apps/claude/protocol/run-controller.ts [hunk 2]
```diff
@@ -2356,6 +2356,16 @@ export function createChatController(deps: ControllerDeps): ChatController {
...
+      if (res.deleted) {
+        // A PRESS ON A ROW THAT NO LONGER EXISTS. The server used to answer
+        // this with the same empty payload a not-yet-written chat gets, and
+        // the page opened a blank conversation with nothing to say why
+        // (Akshil, 2026-09-15). Not cached: the answer is about a task that is
+        // gone, and a cache entry for it would outlive the listing's memory.
+        landHistory(res, false);
+        addError("This task was deleted. Its conversation is gone; pick another chat or start a new one.");
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (AkshilVT) [reply frontend/src/apps/claude/protocol/run-controller.ts]: Fixed in c22f4955f.
